app/models/dispatcher.rb

```diff
@@ -99,10 +96,16 @@ def on_comment_created(comment)
     end
   end
 
-  def on_proposal_update(_proposal)
+  def on_proposal_update(modifier)
+    if proposal.client_slug == "ncr"
```

Comment:
we were delegating  before and it was getting very confusing ->

`Dispatcher` would initialize `NcrDispatcher` or `LinearDispatcher`, `NcrDispatcher` inherited from `LinearDispatcher`, which in turn inherited from `Dispatcher`. This PR is an attempt to make the branching much more explicit. I expect that we will have separate dispatchers at some point in the future, but consolidating makes the logic easier to read for now and then we can break out the differences from here.

Why is `proposal.ncr?` less brittle than looking at the `client_slug` attribute? I am open to it, just wondering.